### DISCUSSION TITLE
Any way to store remote authentication?

### DISCUSSION THREAD

**Discussion by pantsmanuk** [jesseduffield/lazygit#3610] [Q&A] [answered]
Is there a way in LG to store the username/password for a remote?

I have a local OneDev instance on a NAS that I use as a "mirror" repository, and it would be handy if I could keep the username/password "in LG" so I don't have to type the password in every time (the PW was generated with LastPass; I can type it in, but having it "pre-staged" would be super nice).

**Comment by mark2185** [comment, marked as answer]
Not by `lazygit` directly, but `git` can cache it for you and `lazygit` can take advantage of that.

Try using the credential helper and giving it a cache of `3600s`:
```
git config credential.helper 'cache --timeout=3600'
```

Other than that I'd suggest using the remote over `ssh` instead of `https`.

**Reply by pantsmanuk** [reply]
Very many thanks @mark2185 . I'll bump up the global cache timeout to "a sensible number" (an hour or two at least) and see how I get on.

Update: That's entirely perfect for my situation. Many thanks.
